crates/ty_python_semantic/src/types.rs

```diff
@@ -4273,9 +4308,14 @@ impl<'db> Type<'db> {
             .place
         {
             Place::Type(dunder_callable, boundness) => {
+                let inferred_return_ty = || {
+                    dunder_callable
+                        .inferred_return_type(db)
+                        .unwrap_or(Type::unknown())
+                };
                 let bindings = dunder_callable
                     .bindings(db)
-                    .match_parameters(argument_types)
+                    .match_parameters(argument_types, inferred_return_ty)
```

Comment:
naming nit: functions are verbs (and we prefer `type` over `ty`), let's name this callback `infer_return_type` not `inferred_return_ty`